src/api/commands/proposal/ProposalPostCommand.ts

```diff
@@ -88,21 +88,21 @@ export class ProposalPostCommand extends BaseCommand implements RpcCommandInterf
         let market: resources.Market;
         const marketModel = await this.marketService.getDefault();
         if (!marketModel) {
-            throw new MessageException(`Default market doesn't exist!`);
+            this.log.warn('Default market doesn\'t exist!');
+            throw new MessageException('Default market doesn\'t exist!');
         }
         market = marketModel.toJSON();
 
         // rest of the data.params are option descriptions
         const optionsList: string[] = data.params;
 
         // todo: get rid of the blocks
-        const daysRetention = Math.ceil((blockEnd - blockStart) / (24 * 30));
-        return await this.proposalActionService.send(proposalTitle, proposalDescription, blockStart, blockEnd,
+        return await this.proposalActionService.send(proposalTitle, proposalDescription,
```

Comment:
the command parameter daysRetention means the amount of days the user wants this proposal to be live, starting from now. at this point, to make things simpler, we're not allowing the user to set a timeStart in the future. proposal voting starts now and ends in now+daysRetention.

add timeStart, timeEnd.

timeStart = now
timeEnd = now + daysRetention